### PR TITLE
[KYUUBI #5529][AUTHZ] Support create table command for Delta Lake

### LINKED ISSUE #5529: [Subtask] Support create table command for Delta Lake
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Support create table command for Delta Lake in Authz plugin.

### Parent issue

#5477

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zml1206**
### _Why are the changes needed?_
To close #5529.
Support create table command for Delta Lake in Authz. 
https://docs.delta.io/latest/delta-batch.html#create-a-table

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No.

### CODE REVIEW

**Comment by pan3793** [comment]
Nice! cc @bowenliang123 and @AngersZhuuuu

**Comment by bowenliang123** [comment]
Cool, let's bring Ranger authorisation to Delta in Kyuubi Authz !

**Review by bowenliang123 — APPROVED** [review]
LGTM.
Would be more clare if we reorder the test cases in the order for the non-partitioned table, the partitioned table, and then the create/replace table. Or just separated into separated ut.

**Comment by cfmcgrady** [comment]
delta lake also supports creating a table at a path

```sql
-- Create or replace table with path
CREATE OR REPLACE TABLE delta.`/tmp/delta/people10m` (
  id INT,
  firstName STRING,
  middleName STRING,
  lastName STRING,
  gender STRING,
  birthDate TIMESTAMP,
  ssn STRING,
  salary INT
) USING DELTA
```

Shall we also add UTs for this case?

**Comment by bowenliang123** [comment]
Path-based tables are supported in Authz now. No path based policies (eg. HDFS service def) of Ranger are supported.

**Comment by zml1206** [comment]
> LGTM. Would be more clare if we reorder the test cases in the order for the non-partitioned table, the partitioned table, and then the create/replace table. Or just separated into separated ut.

Thanks, Updated.

**Comment by cfmcgrady** [comment]
> Path-based tables are supported in Authz now. No path based policies (eg. HDFS service def) of Ranger are supported.

Does Delta Lake have an option for the globally disabled path-based table, similar to how `spark.sql.runSQLOnFiles` works? If not, I think we can introduce a configuration in Kyuubi AuthZ to disable this behavior temporarily, prior to the completion of the path-based table feature. WDYT? cc @yaooqinn

**Comment by yaooqinn** [comment]
@AngersZhuuuu is currently working one the path/uri authorization

What's the plan of CREATE OR REPLACE TABLE delta.`/tmp/delta/people10m`

**Comment by zml1206** [comment]
> > Path-based tables are supported in Authz now. No path based policies (eg. HDFS service def) of Ranger are supported.
> 
> Does Delta Lake have an option for the globally disabled path-based table, similar to how `spark.sql.runSQLOnFiles` works? If not, I think we can introduce a configuration in Kyuubi AuthZ to disable this behavior temporarily, prior to the completion of the path-based table feature. WDYT? cc @yaooqinn

https://github.com/delta-io/delta/blob/8639c411890a5c77386f04e2282fcf4caa401eff/spark/src/test/scala/org/apache/spark/sql/delta/DeltaDDLUsingPathSuite.scala#L125
 Delta Lake is also controlled by `spark.sql.runSQLOnFiles`.

**Comment by zml1206** [comment]
> @AngersZhuuuu is currently working one the path/uri authorization
> 
> What's the plan of CREATE OR REPLACE TABLE delta.`/tmp/delta/people10m`
```
== Analyzed Logical Plan ==
ReplaceTable [StructField(id,IntegerType,true)], TableSpec(Map(),Some(DELTA),Map(),None,None,None,false), true
+- ResolvedDBObjectName org.apache.spark.sql.delta.catalog.DeltaCatalog@2f9addd4, [delta, /tmp/delta/people10m]

== Optimized Logical Plan ==
CommandResult AtomicReplaceTable org.apache.spark.sql.delta.catalog.DeltaCatalog@2f9addd4, delta.`/tmp/delta/people10m`, [StructField(id,IntegerType,true)], TableSpec(Map(),Some(DELTA),Map(),None,None,None,false), true, org.apache.spark.sql.execution.datasources.v2.DataSourceV2Strategy$$Lambda$1627/2047805101@71b3ba5a
   +- ReplaceTable [StructField(id,IntegerType,true)], TableSpec(Map(),Some(DELTA),Map(),None,None,None,false), true
      +- ResolvedDBObjectName org.apache.spark.sql.delta.catalog.DeltaCatalog@2f9addd4, [delta, /tmp/delta/people10m]
```

**Comment by cfmcgrady** [comment]
> Delta Lake is also controlled by `spark.sql.runSQLOnFiles`.

thank you for your input. @zml1206

**Review by yaooqinn — APPROVED** [review]

**Review by cfmcgrady — APPROVED** [review]

**Comment by cfmcgrady** [comment]
thanks, merging to master(v1.9.0).

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+           |  gender STRING,
+           |  birthDate TIMESTAMP,
+           |  ssn STRING,
+           |  salary INT
+           |) USING DELTA
+           |""".stripMargin
+      interceptContains[AccessControlException] {
+        doAs(someone, sql(createNonPartitionTableSql))
+      }(s"does not have [create] privilege on [$namespace1/$table1]")
+      doAs(admin, createNonPartitionTableSql)
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:81]: Hi @zml1206, this line has an issue that it does not trigger a sql operation.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+           |  ssn STRING,
+           |  salary INT
+           |)
+           |USING DELTA
+           |PARTITIONED BY (gender)
+           |""".stripMargin
+      interceptContains[AccessControlException] {
+        doAs(someone, sql(createPartitionTableSql))
+      }(s"does not have [create] privilege on [$namespace1/$table2]")
+      doAs(admin, createPartitionTableSql)
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:101]: ditto

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+           |  gender STRING,
+           |  birthDate TIMESTAMP,
+           |  ssn STRING,
+           |  salary INT
+           |) USING DELTA
+           |""".stripMargin
+      interceptContains[AccessControlException] {
+        doAs(someone, sql(createOrReplaceTableSql))
+      }(s"does not have [create] privilege on [$namespace1/$table1]")
+      doAs(admin, createOrReplaceTableSql)
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:124]: ditto

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala (hunk 1 above)
Comment (zml1206) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:81]: Yes, I'll fix it right away, thanks.

Comment (zml1206) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:81]: Fix in #5597

**Pull request closed without merging**